moto/route53/models.py

```diff
@@ -215,9 +216,14 @@ def delete(self, *args, **kwargs):
 
 
 def reverse_domain_name(domain_name):
-    if domain_name.endswith("."):  # normalize without trailing dot
-        domain_name = domain_name[:-1]
-    return ".".join(reversed(domain_name.split(".")))
+    labels = domain_name.split(".")
+    if domain_name.endswith("."):
+        labels.pop()
+        labels = list(reversed(labels))
+        labels.append("")
+    else:
+        labels = reversed(labels)
+    return ".".join(labels)
```

Comment:
Why the rewrite? The logic seems the same as it was before

Reply:
Currently "." in the end is always removed, but it should not be according to the example from the docs.

Reply:
I made some tests in AWS Route53 with and without "." in the end. I didn't see any difference in behaviour. May be I miss something in the example provided in the docs. I reverted the changes. Thanks!